helm/resource_release.go

```diff
@@ -1508,3 +1512,13 @@ func valuesKnown(d *schema.ResourceDiff) bool {
 	}
 	return true
 }
+
+func isLocalChart(chart string, repo string) bool {
+	_, err := os.Stat(chart)
+	_, err1 := os.Stat(repo)
+	_, err2 := url.ParseRequestURI(chart)
+	if os.IsNotExist(err) && os.IsNotExist(err1) && err2 != nil {
```

Comment:
Instead of having this big conditional you could invert the logic in the function to check if the `chart` or `repo` _do_ exist then return early, like this:

```go
func isLocalChart(chart string, repo string) bool {
	if _, err := os.Stat(chart); err == nil {
		return true
	}
	if _, err := os.Stat(repo); err == nil {
		return true
	}
	return false
}
```

Reply:
thank you for this. This covers cases for local charts and OCI charts. the one case that it misses is when the chart is a direct url to a `.tgz` file.
```hcl
resource "helm_release" "example" {
  name  = "redis"
  chart = "https://charts.bitnami.com/bitnami/redis-10.7.16.tgz"
}
```

adding URL parse request on `.chart` covers this case

```go
func isLocalChart(chart string, repo string) bool {
	_, urlerr := url.ParseRequestURI(chart)

	if _, err := os.Stat(chart); err == nil || urlerr == nil {
		return true
	}
	if _, err := os.Stat(repo); err == nil {
		return true
	}

	return false
}
```